nixos/modules/hardware/video/nvidia.nix

```diff
@@ -254,10 +254,21 @@ in
         '';
       };
 
-      open = lib.mkEnableOption ''
-        the open source NVIDIA kernel module
+      open = lib.mkOption {
+        example = true;
+        description = "Whether to enable the open source NVIDIA kernel module.";
+        type = lib.types.bool;
```

Comment:
Since this affect quite a few people, we might want to introduce some guiding message here. If we also accept "unset" here, than an assertion could reproduce the changelog above.

Reply:
You can see how people will get confused here: https://github.com/NixOS/nixos-hardware/issues/1092#issue-2494532889

Reply:
> If we also accept "unset" here, then an assertion could reproduce the changelog above.

That's the original version of this pr, accepting null. But then we would have to cast `nullOr types.bool` to `types.bool` on every place referencing `cfg.open`, or adding a pretty ugly `isOpen` indirection. I would rather prefer improving the error message for all mandatory options as in https://github.com/NixOS/nixpkgs/pull/338362

Reply:
> Since this affect quite a few people, we might want to introduce some guiding message here. If we also accept "unset" here, then an assertion could reproduce the changelog above.

Yep, just helped someone that updated their flake and got this issue. Would love to see a PR to make this message cleaner, at least for the short term.